test/unit/custom_spec.js

```diff
@@ -0,0 +1,99 @@
+import { buildGetDocumentParams } from './test_utils';
+import { DOMCanvasFactory } from '../../src/display/dom_utils';
+import { getDocument } from '../../src/display/api';
+import { isNodeJS } from '../../src/shared/util';
+import { PDFJS } from '../../src/display/global';
+
+function getTopLeftPixel(canvasContext) {
+  var imgData = canvasContext.getImageData(0, 0, 1, 1);
+  var r = imgData.data[0];
+  var g = imgData.data[1];
+  var b = imgData.data[2];
+  var a = imgData.data[3];
+  return { r, g, b, a};
+}
+
+describe('custom canvas rendering', function() {
+  var transparentFileName = 'transparent.pdf';
+  var transparentGetDocumentParams = buildGetDocumentParams(transparentFileName);
+
+  var CanvasFactory;
+  var loadingTask;
+  var pdfDocument, page;
+
+  beforeAll(function(done) {
+    if (isNodeJS()) {
+      PDFJS.pdfjsNext = true;
+      // NOTE: To support running the canvas-related tests in Node.js,
+      // a `NodeCanvasFactory` would need to be added (in test_utils.js).
+    } else {
+      CanvasFactory = new DOMCanvasFactory();
+    }
+    loadingTask = getDocument(transparentGetDocumentParams);
+    loadingTask.promise.then(function(doc) {
+      pdfDocument = doc;
+      return pdfDocument.getPage(1);
+    }).then(function(data) {
+      page = data;
+      done();
+    }).catch(function (reason) {
+      done.fail(reason);
+    });
+  });
+
+  afterAll(function(done) {
+    CanvasFactory = null;
+    done();
+  });
+
+  it('renders to canvas with a default white background', function(done) {
+    if (isNodeJS()) {
+      pending('TODO: Support Canvas testing in Node.js.');
+    }
+    var viewport = page.getViewport(1);
+    var canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    var promiseDone = page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).then(function() {
+      var { r, g, b, a } = getTopLeftPixel(canvasAndCtx.context);
+      CanvasFactory.destroy(canvasAndCtx);
+      expect(r).toEqual(255);
+      expect(g).toEqual(255);
+      expect(b).toEqual(255);
+      expect(a).toEqual(255);
+    });
+    promiseDone.then(function() {
+      done();
+    }).catch(function (reason) {
+      done.fail(reason);
+    });
```

Comment:
Nit: The above can probably be simplified to simply (applies in the test below as well):
```javascript
page.render({
  canvasContext: canvasAndCtx.context,
  viewport,
}).then(function() {
  var { r, g, b, a } = getTopLeftPixel(canvasAndCtx.context);
  CanvasFactory.destroy(canvasAndCtx);
  expect(r).toEqual(255);
  expect(g).toEqual(255);
  expect(b).toEqual(255);
  expect(a).toEqual(255);

  done();
}).catch(function (reason) {
  done.fail(reason);
});
```